dcos_installer/upgrade_win.py

```diff
@@ -0,0 +1,251 @@
+        }
+    }
+    finally
+    {
+        $ErrorActionPreference = $backupErrorActionPreference
+    }
+}
+
+function SetupDirectories() {
+    # available directories
+    $dirs = @(
+        "C:\d2iq",
+        "C:\d2iq\dcos",
+        "C:\d2iq\dcos\etc",
+        "$($basedir)",
+        "$($basedir)\bootstrap",
+        "$($basedir)\bootstrap\prerequisites",
+		"$($basedir)\bin",
+        "$($basedir)\etc",
+        "$($vardir)",
+        "$($vardir)\log"
+    )
+    # setup
+    Write-Log -Level "Info" -LogContent "Creating a directories structure:"
+    foreach ($dir in $dirs) {
+        if (-not (test-path "$dir") ) {
+            Write-Log -Level "Debug" -LogContent "$($dir) doesn't exist, creating it"
+            New-Item -Path $dir -ItemType directory | Out-Null
+        } else {
+            Write-Log -Level "Debug" -LogContent "$($dir) exists, no need to create it"
+        }
+    }
+}
+
+function CreateWriteFile([String] $dir, [String] $file, [String] $content) {
+    Write-Log -Level "Debug" -LogContent "vars: $dir, $file, $content"
+    if (-not (test-path "$($dir)\$($file)") ) {
+        Write-Log -Level "Debug" -LogContent "Creating $($file) at $($dir)"
+    }
+    else {
+        Write-Log -Level "Warn" -LogContent "$($dir)\$($file) already exists. Re-writing"
+        Remove-Item "$($dir)\$($file)"
+    }
+    New-Item -Path "$($dir)\$($file)" -ItemType File
+    Write-Log -Level "Info" -LogContent "Writing content to $($file)"
+    Add-Content "$($dir)\$($file)" "$($content)"
+    Get-Content "$($dir)\$($file)"
+}
+
+
+function main($url, $masters) {
+    SetupDirectories
+
+    # Fill up gen_out arguments to cluster.conf
+    Write-Log -Level "Debug" -LogContent "MASTERS: $($masters)"
+    [System.Array]$masterarray = $masters.replace('"', '').replace('[', '').replace(']', '').replace(' ', '').split(',')
+    $masternodecontent = ""
+    for ($i=0; $i -lt $masterarray.length; $i++) {
+        $masternodecontent += "[master-node-$($i+1)]`nPrivateIPAddr=$($masterarray[$i])`nZookeeperListenerPort=2181`n"
+    }
+    $local_ip = Invoke-NativeApplication {. "$($basedir)\bin\detect_ip.ps1"}
+    Write-Log -Level "Debug" -LogContent "Local IP: $($local_ip)"
+    $content = "$($masternodecontent)`n[distribution-storage]`nRootUrl=$($url)`nPkgRepoPath=windows/packages`nPkgListPath=windows/package_lists/latest.package_list.json`nDcosClusterPkgInfoPath=cluster-package-info.json`n`n[local]`nPrivateIPAddr=$($local_ip)"
+    CreateWriteFile "$($basedir)\etc" "cluster.conf" $content
+
+    Write-Log -Level "Info" -LogContent "Running Winpanda.py upgrade ..."
+    Invoke-NativeApplication {python.exe "$($basedir)\winpanda\bin\winpanda.py" --inst-root-dir="$($basedir)" upgrade}
+    Write-Log -Level "Info" -LogContent "dcos_node_upgrade.ps1 successfully finished."
+}
+
+main $bootstrap_url $masters
+
+"""
```

Comment:
Instead of shortening the lines to < 120 characters, to satisfy tox, you can add `  # noqa` to the end quotes here (note, two spaces, hash, one space, "noqa")

Reply:
thx @jongiddy , will do that and it should help.
Still looking on how to use `master_discovery` instead of `master_list` . From what I see there are few validations for master_discovery being a advanced or simple depends on this.

Reply:
with regard to
1) long lines - added separate template to gen/build_deploy/powershell/dcos_node_upgrade.ps1.in
2) master_discovery - trying workaround by checking the value and define master_list conditionally

Details in "testing PR" https://github.com/dcos/dcos/pull/6913